adebar-cli

```diff
@@ -153,14 +153,15 @@ fi
 # Check whether a device is connected at all and, if configured, the serial matches
 # No device connected:
 ADBOPTS=""
-if [ -z "$(adb devices|egrep "^[0-9A-Za-z.:]+\s+device$"|awk '{print $1}')" ]; then
+
+if [ -z "$(adb devices|grep -E "^[0-9A-Za-z]+[^ ]+device"|awk '{print $1}')" ]; then
```

Comment:
This is a completely different logic: 1) you removed the "end of string" marker, and 2) most likely broke the spacing. An expected string would look like:

    0123456789ABCDEF	device
    012.34:56789ABCD	device

which matches the regex `^[0-9A-Za-z.:]+\s+device$`: {string containing only alpha-numerics, dots and colons} {followed by at least one white space} {followed by the literal term "device"} {end-of-string}. Your `[^ ]`, instead of "white spaces" would match "everything but a blank". It probably works for you right now as there's a "tab" in that place, but it might break for others. You also removed `.:` from the device name, which is definitely needed there (some devices have it; your replacement currently still works, but only if the two are separated by a tab).

What broke there with the original regex? We need to fix that in some other way. Please let's use either this PR's main area or your original issue to figure. Include the output of `adb devices` you get, so I can help out :)